x/evm/types/denom_config.go

```diff
@@ -65,8 +65,8 @@ func GetEVMCoinDenom() string {
 	return evmCoinInfo.Denom
 }
 
-// setEVMCoinInfo allows to define denom and decimals of the coin used in the EVM.
-func setEVMCoinInfo(evmdenom EvmCoinInfo) {
+// SetEVMCoinInfo allows to define denom and decimals of the coin used in the EVM.
+func SetEVMCoinInfo(evmdenom EvmCoinInfo) {
```

Comment:
This method should not be public otherwise it is possible to modify the evm denom in multiple places and multiple times.

Reply:
maybe we can also introduce a `.Seal()` method that raises an error when trying to change the EVM denom multiple times similar to how the SDK config is working